parsl/serialize/facade.py

```diff
@@ -0,0 +1,163 @@
+#  from parsl.serialize.concretes import *
+from parsl.serialize.base import METHODS_MAP_DATA, METHODS_MAP_CODE
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class ParslSerializer(object):
+    """
+    Information that we want to be able to ship around:
+
+
+    * Function run information ->
+    * Function id <---> Container id ? (is there a 1-1 mapping here?)
+    * Container id
+    * Endpoint id
+
+    * Function body
+    * Potentially in a byte compiled form ?
+    * All parameters ->
+    * Args  + Kwargs
+
+
+    From the client side. At function invocation we need to capture
+
+    """
+
+    def __init__(self):
+        """ Instantiate the appropriate classes
+        """
+        headers = list(METHODS_MAP_CODE.keys()) + list(METHODS_MAP_DATA.keys())
+        self.header_size = len(headers[0])
+
+        self.methods_for_code = {}
```

Comment:
It's not clear to me what this code/data distinction is for.

Reply:
If the object is not a callable (data), we only use pickle and dill. However if the object is code pickle/dill may not work straight-away, and we might need to code as text and do some extra handling. Secondly, I want to add caching for serialized code, and not for data.

Reply:
I don't understand either of those distinctions:

i) what about non-callable objects where pickle/dill don't work?
ii) what is special about a callable object that I am sending round that isn't special about a non-callable object that I'm sending around?